src/templates/template-text-classification/main.py

```diff
@@ -25,8 +32,16 @@ def run(local_rank: int, config: Any):
     # create output folder and copy config file to output dir
     config.output_dir = setup_output_dir(config, rank)
     if rank == 0:
+        #::: if ((it.argparser == 'fire')) { :::#
+        with open(f"{config.output_dir}/config-lock.yaml", "a+") as f:
+            for key, value in config.items():
+                f.write(f"{key}: {value}\n")
+
+        #::: } else { :::#
         copy(config.config, f"{config.output_dir}/config-lock.yaml")
 
+        #::: } :::#
```

Comment:
As for python-fire you need to write parsed config into the file, let's do that before adding `config.output_dir`. 
Can you check the diff between saved config and the initial one get the idea if `config-lock.yaml` could be reused in a new training ?

Reply:
@vfdev-5 I tried this and this seems to work ok except for the optional arguments i.e. `train_epoch_length` and `eval_epoch_length`  when these values are None. So we may need to add a None check here

Reply:
Can you provide an example of what you say to make it clear

Reply:
Yeah so 
The config output
```
seed: 666
data_path: ./
train_batch_size: 32
eval_batch_size: 32
num_workers: 4
max_epochs: 20
use_amp: False
debug: False
output_dir: logs/20230815-122747-backend-None-lr-0.0001
log_every_iters: 2
lr: 0.0001
model: resnet18
train_epoch_length: None
eval_epoch_length: None
backend: None
model: resnet18
```
The config input 
```
seed: 666
data_path: ./
train_batch_size: 32
eval_batch_size: 32
num_workers: 4
max_epochs: 20
use_amp: false
debug: false
output_dir: ./logs
log_every_iters: 2
lr: 0.0001
model: resnet18
```